[cryptolib] KDF API changes

ballifatih opened the pull request:
NIST SP 800-108r1 defines multiple iteration modes for given PRF \in {KMAC, HMAC, CMAC}. Our current plan is to support:

- counter mode with HMAC
- (no-mode) with KMAC, since KMAC can process produce arbitrary-size outputs

This PR modifies the API so that these two are separate functions. _Label_ and _Context_ inputs are also added to API.

jadephilipoom commented on sw/device/lib/crypto/include/datatypes.h:
```diff
@@ -320,9 +322,12 @@ typedef enum otcrypto_key_mode {
-  // Key is intended for KDF-CTR with KMAC as PRF.
-  kOtcryptoKeyModeKdfCtrKmac =
-      kOtcryptoKeyTypeKdf << 16 | kOtcryptoKdfKeyModeCtrKmac,
+  // Key is intended for KDF-CTR with KMAC128 as PRF.
```
nit:
```suggestion
  // Key is intended for KDF with KMAC128 as PRF.
```
Same for two lines down.

jadephilipoom reviewed (approved):
Thanks, Fatih! LGTM.

wettermo commented on sw/device/lib/crypto/include/kdf.h:
```diff
@@ -19,22 +20,41 @@ extern "C" {
+ * `checksum` field of the blinded key struct will be populated by this
+ * function.
+ *
+ * @param key_derivation_key Pointer to the blinded key derivation key.
```
For `key_derivation_key` it says "Pointer" in the comment, but below the parameter isn't a pointer type.
Same for `keying_material`.

ballifatih commented:
Thanks for the review @jadephilipoom @wettermo. I made few small post-review changes for the sake of CI:

- Updated `kdf.c` to match the new definitions of the new functions.
- Updated the cryptolib's doc so that one obsolete type is removed and function names are added.

I think it is ready to merge.
